frontend/src/components/Home.test.tsx

```diff
@@ -0,0 +1,38 @@
+import "@testing-library/jest-dom";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { vi } from "vitest";
+
+import Home from "./Home";
+
+describe("Home", () => {
+  test("renders a heading", () => {
+    render(<Home />);
+
+    const heading = screen.getByRole("heading", { level: 1 });
+    const headingText = screen.getByRole("heading", {
+      name: /smartshopper/i,
+    });
+
+    expect(heading).toBeInTheDocument();
+    expect(headingText).toBeInTheDocument();
+  });
+
+  test("renders a button with text 'Click Me!'", () => {
+    render(<Home />);
+
+    const button = screen.getByRole("button", { name: /click me!/i });
+
+    expect(button).toBeInTheDocument();
+  });
+
+  test("calls alert when clicked", () => {
```

Comment:
🐘 Suggested format - just trying to be as descriptive as possible without naming props/specific text and other details that may change even if functionality remains. For example the exact text for the button may change but the purpose is the same.

```suggestion
describe("When the user clicks the only button", () => {
  it("Then displays an alert telling the user the button has been clicked, () => {
```